lib/index.js

```diff
@@ -79,16 +80,50 @@ module.exports = {
                             config.validate.params = config.validate.params || {};
                             config.validate.params[validator.parameter.name] = validator.schema;
                             break;
-                        case 'query':
-                            config.validate.query = config.validate.query || {};
-                            config.validate.query[validator.parameter.name] = validator.schema;
-                            break;
                         case 'body':
                         case 'form':
                             config.validate.payload = validator.schema;
                             break;
                     }
                 });
+
+                config.validate.query = function(value, options, callback) {
+                    var validationErrors = new Array();
+                    for( var i= 0, j=Object.keys(value).length; i<j; i++ ) {
+                        route.validators.forEach(function (validator) {
+                            if( validator.parameter.name === Object.keys(value)[i] ) {
+                                if( validator.parameter.type === 'array' ) {
+                                    validator.validate(value[validator.parameter.name], function(err, result) {
+                                        if( err ) {
+                                            validationErrors.push(err);
+                                        }
+                                        else {
+                                            value[validator.parameter.name] = result;
+                                        }
+                                    });
+                                }
+                                else {
+                                    Joi.validate(value[validator.parameter.name], validator.schema, options, function(err, result) {
```

Comment:
Also more nitpicks - please follow coding conventions. Example:

`if (validator.parameter.name...` and not `if( validator.parameter.name...`.

`for (var i = 0...` and not `for( var i= 0....` and other whitespace issues.